core.ts

```diff
@@ -227,31 +227,17 @@ const isPrimitive = (
 	);
 };
 
-const baseEscapeResource = (resource: string | string[]) => {
+// Escape a resource name (string), or resource path (array)
+const escapeResource = (resource: string | string[]) => {
 	if (isString(resource)) {
-		return encodeURIComponent(resource);
+		resource = encodeURIComponent(resource);
 	} else if (Array.isArray(resource)) {
-		return resource.map(encodeURIComponent).join('/');
+		resource = resource.map(encodeURIComponent).join('/');
 	} else {
 		throw new Error('Not a valid resource: ' + typeof resource);
 	}
-};
-
-// Escape a resource name (string), or resource path (array)
-const escapeResource = (resource: string | string[]) => {
-	const encodedResource = baseEscapeResource(resource);
-
-	if (trailingCountRegex.test(encodedResource)) {
-		throw new Error(
```

Comment:
It basically boils down to removing this restriction/separate function and then in-lining `baseEscapeResource` since we only have `escapeResource` now which allows $count